lib/reduce_helper.cu

```diff
@@ -78,8 +78,12 @@ namespace quda
       }
 
       if (!reduce_count) {
-        reduce_count = static_cast<count_t *>(device_malloc(QUDA_MAX_MULTI_REDUCE * sizeof(decltype(*reduce_count))));
-        init_count<<<1, 1>>>(reduce_count, QUDA_MAX_MULTI_REDUCE);
+        reduce_count = static_cast<count_t *>(device_malloc(max_n_reduce() * sizeof(decltype(*reduce_count))));
+        TuneParam tp;
+        tp.grid = dim3(1, 1, 1);
+        tp.block = dim3(1, 1, 1);
+
+        qudaLaunchKernel(init_count<count_t>, tp, 0, reduce_count);
       }
 
       cudaEventCreateWithFlags(&reduceEnd, cudaEventDisableTiming);
```

Comment:
`qudaEventCreateWithFlags`? Or appropriate abstraction tbd?

Reply:
The streams and events will be dealt with in the next abstraction PR